networking.md

```diff
@@ -0,0 +1,88 @@
+# Filecoin Network
+
+The filecoin network is built using libp2p building blocks, as transports and protocols, as well as some additional filecoin specific protocols as outlined in [Network Protocols](./network-protocols.md).
+
+## Required Protocols
+
+Every full node must support the following libp2p protocols:
+
+ - [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) ([for data announcements](./data-propagation.md))
+ - [bitswap] ([for data exchange](./data-propagation.md))
+ - Filecoin specific Protocols:
+   - Hello Handshake
+   - StorageDeal
+   - BlockSync
+
+## Transports, Streams & Encryption
+
+Connections between nodes are end-to-end-encrypted and -authenticated, thus every node has a `peerId` associated with it.
+
+Filecoin protocols are run as multiple streams over a single connection. Only one connection should exist between two nodes at any time, multiple streams within that connections must be used to exchange different protocols. This is usually handled from within the libp2p stack if configured properly.
+
+This document doesn't make any assumptions on which transports and multiplexing features a particular node offers, however it must at least be able to respond and communicate through mplex channels via a secio transport.
+
+{{% notice info %}}
+Implementers are encouraged to offer and experiment with more protocols and transports, as long as the above mentioned requirements of ee2e, authentication and muxing are still met. In particular the authors encourage to look into TLS1.3 and QUIC as libp2p protocol/transport layers.
+{{% /notice %}}
+
+### Establishing connections
+
+When a filecoin node connects to the network, it may connect to multiple nodes at once. When a new node connects, it must be greeted with the `HelloMessage` through the hello-handshake protocol first. Only after that message has been sent can other streams be opened. If a node receives a `HelloMessage` with a GenesisBlock it doesn't support, it must immediately close the entire connection to that peer.
+
+If the node learns through that handshake about newer blocks it should request these immediately to sync up their chain.
```

Comment:
I think the situation here may actually be a bit more complex: generally, some "newer blocks" may be invalid (not in themselves but in leading back to genesis -- i.e. DoS) or there may be other required checks to be performed when establishing a new connection with a node, vs receiving blocks from an existing connection. I would defer here: replace "it should request these immediately to sync up their chain." with "it should initiate the syncing sequence" and we'll defer details to there (as is currently done anyways)

Reply:
I agree with the stated problems, but am not sure how this is relevant here, but it clearly becomes cloudy with us being so specific here. I'll rephrase the last part to leave it more open to `sync` how that actually takes place. It was for sure not meant to "inject" this block into a running sync process.